File: golang/commonstructs/tree_node.go
package commonstructs

/**
 * Definition for a binary tree node.
 */
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func (node1 *TreeNode) Equals(node2 *TreeNode) bool {
	if node1 == nil && node2 == nil {
		return true
	}

	if node1 == nil || node2 == nil {
		return false
	}

	if node1.Val != node2.Val {
		return false
	}

	equals := node1.Left.Equals(node2.Left)
	if !equals {
		return false
	}

	equals = node1.Right.Equals(node2.Right)

	return equals
}

File: csharp/Solutions/CommonClasses/TreeNode.cs
/// <summary>
/// Definition for a binary tree node.
/// </summary>

namespace Solutions.CommonClasses;

public class TreeNode(int val, TreeNode? left = null, TreeNode? right = null)
{
    public int Val = val;
    public readonly TreeNode? Left = left;
    public readonly TreeNode? Right = right;
}